### DISCUSSION TITLE
isDirty remains to be true even no changes on field

### DISCUSSION THREAD

**Discussion by dileepthomas** [react-hook-form/react-hook-form#6448] [Q&A]
### Version Number

6.14.1

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-usefieldarray-select-cufgz

### Steps to reproduce

1. Go to https://codesandbox.io/s/react-hook-form-usefieldarray-select-cufgz
2. Click on APPEND button
3. See the SUBMIT button and the text isDirty

If we change the version to 6.9.6 its working as expected but not in 6.14.1

### Expected behaviour

isDirty should remain false even though its updated using setValue and the shouldDirty is false for the field

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
have you tried the latest v6?

**Comment by bluebill1049** [comment]
This is expected behavior: when you append a new input, the input value will be updated and hence it will no longer to be matched with your `defaultValues`.

**Comment by dileepthomas** [comment]
Okay, But my doubt is when i click on append and I remove the appended field why the isDiry is true, because I didn't typed anything so won't it should be defaultValues itself  and isDirty shouldn't be false?

**Comment by bluebill1049** [comment]
native input will be an empty string upon the append or prepend action. This is expected behaviour include the latest v7.

**Comment by dileepthomas** [comment]
let me rephrase the workflow so the user logs into the form, since he has already prefilled the username, in useEffect updated the defaultValue of username from empty string to the actual username, and by default 4 array fields will be there. Now as per the user these are defaultValues.

Now the user clicks on the append, now the user changed  the  default form of 1 username and 4 array fields to 1 username and 5 array fields. Now he doesn't type anything and clicks on delete, now it came back to 1 username and 4 array fields which is the default for the user

So at this time also isDirty is true,  based on this isDirty we enable or disable the submit field, but since he came back to the defaultValues state (1 username and 4 array fieldls) as like the page is loaded why the user need to see the Save button as enabled ?

May be I am new to this, but help me in understanding the above if the assumption and flow is wrong ?

**Comment by bluebill1049** [comment]
`dirty` check with field array is always a hard topic, the only way you can revert back to `isDirty` to `false` is if you invoke reset() again with current field values, so basically let hook form refresh its defaultValues so it matches with current formValues, there is no magic, hook form will do a deep equal with current form values and your defaultValues.

If you append, prepend, insert a new row of data, your form will be dirty even user is not interacted with the form, why? because your form values will be updated and they are no longer match with your defaultValues. 

think the following: this is basically append

```tsx
const test = [];

test.push('1')

[] !=== ['1']
```

**Comment by dileepthomas** [comment]
So how should i solve this to achieve the output ?
https://codesandbox.io/s/react-hook-form-usefieldarray-select-cufgz

**Comment by bluebill1049** [comment]
like I suggested above, reset the entire `formValues` with the current `formValues`.

**Comment by dileepthomas** [comment]
after removing the index i am doing the reset, but its showing only username field, if you click on append and delete you can see. Did I missed anything
https://codesandbox.io/s/react-hook-form-usefieldarray-select-cufgz

**Comment by bluebill1049** [comment]
at `useEffect`

https://codesandbox.io/s/react-hook-form-usefieldarray-select-forked-kuuur?file=/src/index.js
